crates/astria-composer/src/executor/bundle_factory/mod.rs

```diff
@@ -189,10 +240,17 @@ impl BundleFactory {
     ///
     /// Returns an empty bundle if there are no bundled transactions.
     pub(super) fn pop_now(&mut self) -> SizedBundle {
-        self.finished
+        let bundle = self
+            .finished
             .pop_front()
             .or_else(|| Some(self.curr_bundle.flush()))
-            .unwrap_or(SizedBundle::new(self.curr_bundle.max_size))
+            .unwrap_or(SizedBundle::new(self.curr_bundle.max_size));
+        self.full = false;
+        bundle
+    }
+
+    pub(super) fn full(&self) -> bool {
+        self.full
```

Comment:
can we just use `self.finished.len() >= self.finished_queue_capacity` here? rather than maintaining a seperate `self.full` variable?

Reply:
The `BundleFactory` can have a full `finished` queue but still have space in its `curr_bundle`.